layered_store/src/store.rs

```diff
@@ -42,6 +42,7 @@ pub struct LayeredStore<Backend: StoreBackend> {
     backend: Backend,
     pending_state: HashMap<String, Option<(DynStoreType, StoreTypeSer)>>,
     modified_keys: HashSet<String>,
+    tmp_state: HashMap<String, Option<(DynStoreType, StoreTypeSer)>>,
```

Comment:
I'd just reuse `pending_state`, the only difference is that for tmp values you do not update `modified_keys`.
You can also implement that with existing `get`/`set` methods by adding flag